File: rb/src/day01.rb
require_relative 'util'

def part1(input)
  ans = input
          .lines
          .map(&:to_i)
          .each_cons(2)
          .filter{_2 > _1}
          .count
  puts ans
end

def part2(input)
  nums = input.lines.map(&:to_i)
  ans = nums
          .each_cons(3)
          .each_cons(2)
          .filter{_2.sum > _1.sum}
          .count
  puts ans
end

with_input do |input|
  part1(input)
  part2(input)
end

File: ts/src/day01.ts
import * as util from "./util"

function part1(input: string) {
  let ans: number = 0
  let last = Number.MAX_VALUE
  input.split(/\n/).map(Number).map(n => {
    if (n > last) {
      ans++
    }
    last = n
  })

  console.log(ans)
}

function part2(input: string) {
  let ans: number = 0
  let last = Number.MAX_VALUE
  input.split(/\n/)
    .map(Number)
    .each_cons(3,
      (...nums) => {
        const next = util.sum(nums)
        if (next > last) {
          ans++
        }
        last = next
      })

  console.log(ans)
}

util.withInput("input/day01.txt", input => {
  part1(input)
  part2(input)
})